src/open_inwoner/plans/forms.py

```diff
@@ -13,6 +14,104 @@
 from .models import Plan, PlanTemplate
 
 
+class PlanTemplateChoiceForm(forms.Form):
+    template = forms.ModelChoiceField(
+        queryset=PlanTemplate.objects.all(),
+        required=True,
+        empty_label=_("No template"),
+        widget=forms.widgets.HiddenInput(),  # Form is used for validation only
+    )
+
+
+class CreatePlanFromTemplateForm(forms.ModelForm):
+    class Meta:
+        model = Plan
+        fields = (
+            "end_date",
+            "plan_contacts",
+        )
+
+    def __init__(self, user, template, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.user = user
+        self.template = template
+        user_contacts = self.user.get_active_contacts()
+        self.fields["plan_contacts"].queryset = user_contacts
+
+        # NOTE we have to convert the ID value of the choice to string to make components recognize checked (taiga 899)
+        self.fields["plan_contacts"].choices = [
+            [str(c.id), c.get_full_name()] for c in user_contacts
+        ]
+
+    def clean_plan_contacts(self):
+        # Make sure current user exists in plan_contacts when editing form
+        data = self.cleaned_data["plan_contacts"]
+        if self.instance.pk:
+            data |= User.objects.filter(pk=self.user.pk)
+        return data.distinct()
+
+    def clean(self) -> dict[str, Any]:
+        cleaned_data = super().clean()
+        end_date = cleaned_data.get("end_date")
+        if end_date:
```

Comment:
You could do:

```
        if (
            (end_date := cleaned_data.get("end_date"))
            and (action_templates := self.template.action_template.all())
        ):
            latest_end_in_days = max([a.end_in_days for a in actionTemplates])
```
to reduce nesting, though I'm  not sure if this is cleaner in this case